packages/hash/frontend/src/blocks/page/BlockContext.tsx

```diff
@@ -0,0 +1,24 @@
+import React, { useContext } from "react";
+
+type BlockContextType = {
+  id: string;
+  error: boolean;
+  setError: (error: boolean) => void;
+};
+
+/** used to detect whether or not a context value was provided */
+const nullBlockView = {};
+
+export const BlockContext = React.createContext<BlockContextType>(
+  nullBlockView as BlockContextType,
+);
+
+export const useBlockContext = () => {
+  const blockContext = useContext(BlockContext);
+
+  if (blockContext === nullBlockView) {
+    throw new Error("no BlockViewContext value has been provided");
+  }
+
+  return blockContext;
+};
```

Comment:
Generally a cleaner way to sort this is to make the default value for the context null, and then check for that in the hook. 

```suggestion
export const BlockContext = React.createContext<BlockContextType | null>(null);

export const useBlockContext = () => {
  const blockContext = useContext(BlockContext);

  if (!blockContext) {
    throw new Error("no BlockViewContext value has been provided");
  }

  return blockContext;
};
```